### ISSUE TITLE
For tf.keras models of non-trivial topology, the keras tag is unavailable in the Graph Dashboard

### ISSUE THREAD

**Issue by caisq** [tensorflow/tensorboard#1991]
Version: tensorflow 2.0.0a0.

Python code to reproduce the error:

```py
import numpy as np
import tensorflow as tf

print(tf.__version__)

input1 = tf.keras.Input(shape=[4])
input2 = tf.keras.Input(shape=[4])
input3 = tf.keras.Input(shape=[4])
dense_layer = tf.keras.layers.Dense(1)

intermediate1 = dense_layer(input1)
intermediate2 = dense_layer(input2)
intermediate3 = dense_layer(input3)

output1 = tf.keras.layers.add([intermediate1, intermediate2])

dense_2_layer = tf.keras.layers.Dense(1)
output2 = dense_2_layer(intermediate1)

model = tf.keras.Model(inputs=[input1, input2, input3],
                       outputs=[output1, output2])
model.compile(loss=['mean_squared_error', 'mean_squared_error'],
              optimizer='sgd')
model.summary()

num_examples = 8
xs1 = np.zeros([num_examples, 4])
xs2 = np.ones([num_examples, 4])
xs3 = np.ones([num_examples, 4])
ys1 = np.ones([num_examples, 1])
ys2 = np.zeros([num_examples, 1])

model.fit([xs1, xs2, xs3], [ys1, ys2], epochs=10,
          callbacks=[tf.keras.callbacks.TensorBoard('/tmp/foo4')])
```

Then open tensorboard:

```sh
tensorboard --logdir /tmp/foo4
```

Open the Graph dashboard and observe that there is no "keras" tag there. 
For models of simpler topology, e.g., a sequential keras model, the tag is available.

**Comment by stephanwlee** [comment]
Hi @caisq, thanks for raising the issue and the helpful code.

 When I looked into a bit, it looks like Keras `model.to_json` raises an exception with the model. Upon more inspection, I realized that the `input3` or `intermdaite3` is not used in any meaningful way and because you specify `input3` in the `inputs` to the Model, the model fails to serialize properly.

Now, I think Keras can improve the error message and a little bit to be a little bit more helpful with its error message and how they handle this type of serialization. I filed a bug (b/128314124) with smaller repro case on them.
